src/Cli/dotnet/Program.cs

```diff
@@ -204,34 +225,52 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient = null
                 Console.WriteLine($"Telemetry is: {(telemetryClient.Enabled ? "Enabled" : "Disabled")}");
             }
 
+            PerformanceLogEventSource.Log.TelemetrySendIfEnabledStart();
             TelemetryEventEntry.SendFiltered(topLevelCommandParserResult);
+            PerformanceLogEventSource.Log.TelemetrySendIfEnabledStop();
 
             int exitCode;
             if (BuiltInCommandsCatalog.Commands.TryGetValue(topLevelCommandParserResult.Command, out var builtIn))
             {
+                PerformanceLogEventSource.Log.BuiltInCommandParserStart();
                 var parseResult = Parser.Instance.ParseFrom($"dotnet {topLevelCommandParserResult.Command}", appArgs.ToArray());
+                PerformanceLogEventSource.Log.BuiltInCommandParserStop();
+
                 if (!parseResult.Errors.Any())
                 {
+                    PerformanceLogEventSource.Log.TelemetrySendIfEnabledStart();
                     TelemetryEventEntry.SendFiltered(parseResult);
+                    PerformanceLogEventSource.Log.TelemetrySendIfEnabledStop();
                 }
 
+                PerformanceLogEventSource.Log.BuiltInCommandStart();
                 exitCode = builtIn.Command(appArgs.ToArray());
+                PerformanceLogEventSource.Log.BuiltInCommandStop();
             }
             else if (string.IsNullOrEmpty(topLevelCommandParserResult.Command))
             {
                 exitCode = 0;
             }
             else
             {
-                CommandResult result = CommandFactoryUsingResolver.Create(
+                PerformanceLogEventSource.Log.ExtensibleCommandResolverStart();
+                Command resolvedCommand = CommandFactoryUsingResolver.Create(
                         "dotnet-" + topLevelCommandParserResult.Command,
                         appArgs,
-                        FrameworkConstants.CommonFrameworks.NetStandardApp15)
-                    .Execute();
+                        FrameworkConstants.CommonFrameworks.NetStandardApp15);
+                PerformanceLogEventSource.Log.ExtensibleCommandResolverStop();
+
+                PerformanceLogEventSource.Log.ExtensibleCommandStart();
+                CommandResult result = resolvedCommand.Execute();
+                PerformanceLogEventSource.Log.ExtensibleCommandStop();
+                
                 exitCode = result.ExitCode;
             }
 
+            PerformanceLogEventSource.Log.TelemetryClientFlushStart();
```

Comment:
telemetry flush constantly in the back ground. This flush is to ensure it is empty before exiting

Reply:
Thanks for the info.  My goal for these events is to make sure that we aren't holding the process open for longer than necessary due to issues where telemetry takes longer to send then we'd like.

---
In reply to: [505716128](https://github.com/wli3/sdk/pull/1#discussion_r505716128) [](ancestors = 505716128)